### PR TITLE
Feat: Local sort and pagination, pus api and visual updates

### PR DESCRIPTION
**Pull request by muselesscreator**
Updates the api to match update frontend contracts doc.
Modifies incoming courses to locally uniquely identify, sort, and paginate them.
a number of small UI updates to account for api updates
Also adds fixes from design review re: email confirmation and entitlement modals, user dropdown, sidebar behavior, and select session modal text and style

https://user-images.githubusercontent.com/5533134/183706988-2450bb9c-7f83-4b2f-a5f6-924965a0cb9d.mov

JIRA:
[AU-688](https://2u-internal.atlassian.net/browse/AU-688)
[AU-748](https://2u-internal.atlassian.net/browse/AU-748)

### CODE REVIEW

**Review by mattcarter — APPROVED** [review]
approved

File: src/data/redux/app/selectors.js [hunk 1]
```diff
@@ -10,24 +11,36 @@ const mkSimpleSelector = (cb) => createSelector([module.appSelector], cb);
...
+export const courseCardData = (state, cardId) => (
+  module.simpleSelectors.courseData(state)[cardId]
 );
 
+const mkCardSelector = (sel) => (state, cardId) => {
+  const cardData = module.courseCardData(state, cardId);
+  if (cardData) {
+    return sel(cardData);
+  }
+  return {};
```
Comment (leangseu-edx) [inline src/data/redux/app/selectors.js:41]: I do not like that this has default return. Most of the time we encourage `StrictDict` and this is arguably the opposite of that.

File: src/App.scss [hunk 2]
```diff
@@ -12,6 +12,9 @@ $input-focus-box-shadow: $input-box-shadow; // hack to get upgrade to paragon 4.
 @import "~@edx/frontend-component-footer/dist/_footer";
 
 #root {
+  input[type=checkbox] {
```
Comment (leangseu-edx) [inline src/App.scss:15]: This feel a bit of too general of css change. If this is for filter, we should move it into the subcomponent and subclass.

File: src/data/redux/app/selectors.js (hunk 1 above)
Comment (muselesscreator) [reply src/data/redux/app/selectors.js:41]: The push for StrictDict is to enforce valid key usage and make sure those bugs don't get dropped. This is making sure that consuming systems get reasonable defaults.

File: src/App.scss (hunk 2 above)
Comment (muselesscreator) [reply src/App.scss:15]: This is to address something that is hitting paragon in general right now.  browser update, we think.  checkbox inputs are all scaled up

**Pull request merged**